mlir/lib/Conversion/VectorToLLVM/ConvertVectorToLLVM.cpp

```diff
@@ -1276,7 +1276,7 @@ struct VectorScalableInsertOpLowering
   matchAndRewrite(vector::ScalableInsertOp insOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     rewriter.replaceOpWithNewOp<LLVM::vector_insert>(
-        insOp, adaptor.getSource(), adaptor.getDest(), adaptor.getPos());
+        insOp, adaptor.getDest(), adaptor.getSource(), adaptor.getPos());
```

Comment:
Can you also add a test for this fix?

Reply:
I'm not sure what extra test I could add? This is tested via the intrinsic being correctly imported/exported to LLVM. If the order is wrong an assert is hit in the generated import/export logic. This is just the one place in MLIR where the builder for the intrinsic is used.

Reply:
This is the lowering from vector to LLVM dialect (convert-vector-to-llvm) and is not related to LLVM IR.

Reply:
There's no bug in the pattern here, I just had to change the order of the operands in the `LLVM::vector_insert` intrinsic to match the equivalent LLVM IR intrinsic. Since if I didn't do that the generated import/export code would fail. 
I then had to make sure all users of the intrinsic also changed the order of operands in the builder calls. This is the only user (as only `vector.scalable.insert` lowers to this intrinsic). 

Note: `LLVM::vector_insert` is the intrinsic (https://llvm.org/docs/LangRef.html#llvm-vector-insert-intrinsic), not to be confused with the insertelement instruction (https://llvm.org/docs/LangRef.html#insertelement-instruction).

Reply:
Oh, I understand now. I did not realize that this change didn't result in a different intrinsic print. Sorry for the noise. In that case, ignore my suggestion 😅

Reply:
No worries :)

Reply:
(I probably should have made it clear that this intrinsic has a custom `assemblyFormat` :sweat_smile:)